server/proxy/proxy.go

```diff
@@ -241,6 +248,9 @@ func HandleUserTcpConnection(pxy Proxy, userConn net.Conn, serverCfg config.Serv
 	xl.Debug("join connections, workConn(l[%s] r[%s]) userConn(l[%s] r[%s])", workConn.LocalAddr().String(),
 		workConn.RemoteAddr().String(), userConn.LocalAddr().String(), userConn.RemoteAddr().String())
 
+	// Record remote address for the access proxy
+	pxy.GetPluginManager().NewUserConn(plugin.CreateNewUserConn(pxy.GetName(), pxy.GetConf().GetBaseInfo().ProxyType, userConn.RemoteAddr()))
```

Comment:
Reference to `NewProxy` operation. We should handle plugin response to decide if we should reject the connection.